octodns/record/__init__.py

```diff
@@ -219,6 +220,18 @@ def changes(self, other, target):
         if self.ttl != other.ttl:
             return Update(self, other)
 
+    def copy(self, zone=None):
+        _type = getattr(self, '_type')
+        if not self._raw_data.get('type'):
+            self._raw_data['type'] = _type
+
+        return Record.new(
+            zone if zone else self.zone,
+            self.name,
+            self._raw_data,
```

Comment:
I believe it'll be safer to do `self.data` here rather than pipe the raw data through so that we get a copy of the record as it is now and not in the state it was when created.

```suggestion
            self.data,
```

Reply:
The issue is that `self.data` only contains the record TTL, not the record type. Sometimes the record object is not a specific record like `ARecord` or `TXTRecord`, but just a `Record` object which has no `_type` attribute.

Reply:
Would it work to do something like?

```python
data = self.data
data['type'] = self._type
# then use `data` as the param.
```

Reply:
All tests passed so everything seems to be OK.